enterprise-suite/scripts/lbc.py

```diff
@@ -0,0 +1,404 @@
+
+def is_running_minishift():
+    stdout, returncode = run('minishift status')
+    return returncode == 0 \
+        and 'minishift: Running' in stdout \
+        and 'openshift: Running' in stdout
+
+# Helm check is a separate function because we also need it when not doing full
+# preflight check, eg. when using --export-yaml argument
+def check_helm():
+    require_version('helm version --client --short', REQ_VER_HELM)
+
+# Kubectl check is needed both in install and verify subcommands
+def check_kubectl(minishift=False):
+    require_version('kubectl version --client=true --short=true', REQ_VER_KUBECTL)
+
+    # Check if kubectl is connected to a cluster. If not connected, version query will timeout.
+    stdout, returncode = run('kubectl version', DEFAULT_TIMEOUT)
+    if returncode != 0:
+        msg = 'Cannot reach cluster with kubectl'
+        if minishift:
+            # Minishift needs special configuration for kubectl to work
+            msg = msg + ". Did you do 'eval $(minishift oc-env)'?"
+        sys.exit(msg)
+
+def preinstall_check(creds, minikube=False, minishift=False):
+    assert minikube == False or minishift == False, 'Did not expect both minikube and minishift running'
+
+    check_helm()
+    check_kubectl()
+
+    require_version("docker version -f '{{.Client.Version}}'", REQ_VER_DOCKER)
+
+    if minikube:
+        require_version('minikube version', REQ_VER_MINIKUBE)
+
+    if minishift:
+        require_version('minishift version', REQ_VER_MINISHIFT)
+        require_version('oc version', REQ_VER_OC)
+
+    if minikube or minishift:
+        # Check if docker is pointing to a cluster
+        if os.environ.get('DOCKER_HOST') == None:
+            sys.exit('Docker CLI is not pointing to a cluster. Did you run "eval $({} docker-env)"?'
+                     .format('minikube' if minikube else 'minishift'))
+
+    # Check if helm is set up inside a cluster
+    stdout, returncode = run('helm version', DEFAULT_TIMEOUT)
+    if returncode != 0:
+        sys.exit('Cannot get helm status. Did you set up helm inside your cluster?')
+
+    # TODO: Check if RBAC rules for tiller are set up
+
+    # Check credentials
+    registry = 'lightbend-docker-commercial-registry.bintray.io'
+    stdout, returncode = run('docker login -u {} --password-stdin {}'.format(creds[0], registry),
+                             6, creds[1])
+    if returncode != 0:
+        sys.exit('Unable to login to lightbend docker registry. Please check your credentials.')
+    else:
+        run('docker logout ' + registry, DEFAULT_TIMEOUT)
+    
+    # TODO: Try to pull docker image from lighbend registry
+    
+    # At the moment when I try 'docker pull {registry}/enterprise-suite/es-monitor-api:latest' I get:
+    # Error response from daemon: Get https://lightbend-commercial-registry.bintray.io/v2/: 
+    # x509: certificate is valid for *.bintray.com, bintray.com, not lightbend-commercial-registry.bintray.io
```

Comment:
Bintray is broken?

Reply:
Not relevant anymore, authentication uses Docker API now